src/metabase/api/embed.clj

```diff
@@ -159,8 +159,14 @@
 
 (defn- add-implicit-card-parameters
   "Add template tag parameter information to `card`'s `:parameters`."
-  [card]
-  (update card :parameters concat (card/template-tag-parameters card)))
+  [{:keys [parameters] :as card}]
+  (assoc card :parameters (->> parameters
+                               ;; in 44 we added card.parameters but we didn't migrate template-tags to parameters
+                               ;; because doing such migration is costly.
+                               ;; so there are cards where some parameters in template-tags does not exist in card.parameters
+                               ;; that why we need to keep concat both of them then dedupe by id
+                               (concat (card/template-tag-parameters card))
```

Comment:
if the above comment is still unclear on why we need to concat the dedupe, try reading this slack [thread](https://metaboat.slack.com/archives/C04DAKKDMFT/p1678947611301079?thread_ts=1678887632.903009&cid=C04DAKKDMFT). Hopefully it gives more context.